pkg/controller/elasticsearch/elasticsearch_controller.go

```diff
@@ -295,6 +310,30 @@ func (r *ReconcileElasticsearch) updateStatus(
 	return common.UpdateStatus(r.Client, cluster)
 }
 
+func (r *ReconcileElasticsearch) annotateResource(
+	ctx context.Context,
+	es esv1.Elasticsearch,
+	reconcileState *esreconcile.State,
+) error {
+	span, _ := apm.StartSpan(ctx, "update_annotations", tracing.SpanTypeApp)
+	defer span.End()
+
+	// cluster-uuid is a special case that is not treated here but through an immediate update due to the risk of data loss
+	// see bootstrap package.
+	newAnnotations, err := reconcileState.Annotations()
+	if err != nil {
+		return err
+	}
+
+	expected := maps.Merge(es.ObjectMeta.DeepCopy().Annotations, newAnnotations)
+	if reflect.DeepEqual(expected, es.Annotations) {
```

Comment:
It does not really matter for now since there is only one hint, however I think we assume here the serialization of `OrchestrationsHints` to be the same across reconciliations. However [I'm not sure there is such a guarantee](https://stackoverflow.com/questions/44755089/does-serialized-content-strictly-follow-the-order-in-definition-use-encoding-json-package).